src/explorer/SiteTreeItem.ts

```diff
@@ -124,6 +136,33 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
             });
     }
 
+    private async enableScmDoBuildDuringDeploy(fsPath: string, runtime: string): Promise<void> {
+        const yesButton: MessageItem = { title: 'Yes' };
+        const noButton: MessageItem = { title: 'No', isCloseAffordance: true };
+        const learnMoreButton: MessageItem = { title: 'Learn More' };
+        const zipIgnoreFolders: string[] = constants.getIgnoredFoldersForDeployment(runtime);
+        const buildDuringDeploy: string = `Run build script during deployment?  The "${zipIgnoreFolders.join(", ")}" directory will be built during the deployment, rather than zipped, resulting in a faster deployment.`;
+        let input: MessageItem = learnMoreButton;
+        while (input === learnMoreButton) {
+            input = await window.showInformationMessage(buildDuringDeploy, yesButton, noButton, learnMoreButton);
+            if (input === learnMoreButton) {
+                // tslint:disable-next-line:no-unsafe-any
+                opn('https://aka.ms/Kwwkbd');
+            }
+        }
+        if (input === yesButton) {
+            let oldSettings: string[] | string = workspace.getConfiguration('appService', Uri.file(fsPath)).get(constants.configurationSettings.zipIgnorePattern);
+            if (typeof oldSettings === "string") {
+                oldSettings = [oldSettings];
+                // settings have to be an array to concat the proper zipIgnoreFolders
+            }
+            workspace.getConfiguration('appService', Uri.file(fsPath)).update(constants.configurationSettings.zipIgnorePattern, oldSettings.concat(zipIgnoreFolders), ConfigurationTarget.WorkspaceFolder);
+            await fse.writeFile(path.join(fsPath, constants.deploymentFileName), constants.deploymentFile);
+        } else {
+            workspace.getConfiguration('appService', Uri.file(fsPath)).update(constants.configurationSettings.neverPromptBuildDuringDeploy, true, ConfigurationTarget.WorkspaceFolder);
```

Comment:
Only update this if they click "Don't warn again". If they close the dialog, we should warn again